spec/lucky/error_handling_spec.cr

```diff
@@ -46,6 +49,21 @@ describe Lucky::ErrorHandler do
     context.response.status_code.should eq(500)
   end
 
+  it "returns a 422 status for InvalidParam exceptions" do
```

Comment:
I'd actually like to keep this one because we specially saying InvalidParam throws a 422, but I like the idea of adding a new test for that case. I've added a second case that tests for a custom response

Reply:
I didn't explain what I meant! I think we should still test InvalidParam, but we can unit test it in a `invalid_param_spec` that checks that the method returns a `422`.

Does that make sense? I don't think we need to go through the error handler to test every error that implements the module. We can test it in a generic fashion and then unit test errors

Reply:
For example:

```crystal
describe InvalidParamError do
it "responds with a custom HTTP code" do
  error = InvalidParamError.new()
  

  error.should be_a(Lucky::HTTPRespondable)
  error.http_code.should eq(422)
end
end
```

Reply:
Oh, I see what you're saying. Ok. I'll move that one then.